Library/Homebrew/system_command.rb

```diff
@@ -128,7 +128,12 @@ def command
 
   attr_reader :executable, :args, :input, :chdir, :env
 
-  attr_predicate :sudo?, :print_stdout?, :print_stderr?, :must_succeed?
+  attr_predicate :print_stdout?, :print_stderr?, :must_succeed?
+
+  sig { returns(T::Boolean) }
+  def sudo?
+    @sudo != false && @sudo != ""
```

Comment:
Similarly: pretty weird and feels like it'd be better to not overload `sudo` in this way